Assets/LeapMotion/Scripts/HandPool.cs

```diff
@@ -59,35 +84,63 @@ public class ModelGroup {
      */
+
     public override HandRepresentation MakeHandRepresentation(Hand hand, ModelType modelType) {
-      HandRepresentation handRep = null;
+      //HandRepresentation handRep = null;
       List<IHandModel> models = new List<IHandModel>();
-      foreach (ModelGroup group in ModelPool) {
-        for (int i = 0; i < group.modelList.Count; i++) {
-          IHandModel model = group.modelList[i];
-          bool isCorrectHandedness;
-          Chirality handChirality = hand.IsRight ? Chirality.Right : Chirality.Left;
-          isCorrectHandedness = model.Handedness == handChirality;
-          if (!EnforceHandedness || model.Handedness == Chirality.Either) {
-            isCorrectHandedness = true;
-          }
-          bool isCorrectModelType;
-          isCorrectModelType = model.HandModelType == modelType;
-          if (isCorrectModelType && isCorrectHandedness) {
-            group.modelList.RemoveAt(i);
-            //--i;
-            models.Add(model);
-            break;
+      Chirality handChirality = hand.IsRight ? Chirality.Right : Chirality.Left;
+      HandRepresentation handRep = new HandProxy(this, hand, handChirality, modelType);
+      for (int i = 0; i < ModelPool.Count; i++) {
+        ModelGroup group = ModelPool[i];
+        if (group.IsEnabled) {
+          IHandModel model = group.TryGetModel(handChirality, modelType);
+          if (model != null) {
+            handRep.AddModel(model);
+            group.modelsCheckedOut.Add(model);
           }
         }
       }
-      handRep = new HandProxy(this, models, hand);
+      ActiveHandReps.Add(handRep);
       return handRep;
     }
-
-    public void ReturnToPool(IHandModel model){
+    public void EnableGroup(string groupName) {
+      for (int i = 0; i < ModelPool.Count; i++) {
+        if (ModelPool[i].GroupName == groupName) {
+          ModelGroup group = ModelPool[i];
+          for (int hp = 0; hp < ActiveHandReps.Count; hp++ ) {
+            HandRepresentation handRep = ActiveHandReps[hp];
+            IHandModel model = group.TryGetModel(handRep.RepChirality, handRep.RepType);
+            if (model != null) {
+              handRep.AddModel(model);
+              group.modelsCheckedOut.Add(model);
+            }
+          }
+          group.IsEnabled = true;
+        }
+      }
+    }
+    public void DisableGroup( string groupName) {
+      for (int i = 0; i < ModelPool.Count; i++) {
+        ModelGroup group = null;
+        if (ModelPool[i].GroupName == groupName) {
+          group = ModelPool[i];
+          for (int m = 0; m < group.modelsCheckedOut.Count; m++) {
+            IHandModel model = group.modelsCheckedOut[m];
+            HandProxy handProxy;
+            if (ModelToHandProxyMapping.TryGetValue(model, out handProxy)) {
+              handProxy.RemoveModel(model);
+            }
+          }
+          group.modelsCheckedOut = new List<IHandModel>();
```

Comment:
Should clear the list here instead of creating a new one from scratch.  Optionally a better solution might be to have a method ReturnToGroup inside of ModelGroup that returns a model to the group.  This method would take care of removing the model from the checked out list.

This line could be an assert that asserts that checked out list is empty, since it always should be after returning all models to the group.